### PR TITLE
Merge plugin upstream and fix compilation erros

### PR DESCRIPTION
**Pull request by sdubov**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/idea/resources/META-INF/plugin.xml [hunk 1]
```diff
@@ -30,25 +30,34 @@
...
     <fileEditorProvider implementation="com.microsoft.azure.hdinsight.jobs.framework.JobViewEditorProvider" />
-
     <moduleBuilder builderClass="com.microsoft.azure.hdinsight.projects.HDInsightModuleBuilder"/>
     <toolWindow
-        anchor="bottom"
-        factoryClass="com.microsoft.azure.hdinsight.spark.ui.SparkSubmissionToolWindowFactory"
-        id="HDInsight Spark Submission"
-        canCloseContents="false"
-        icon="/icons/SparkSubmissionToolWindow.png"/>
+            anchor="bottom"
```
Comment (maartenba) [inline PluginsAndFeatures/azure-toolkit-for-intellij/idea/resources/META-INF/plugin.xml:34]: Looks like `icon` was lost?

File: PluginsAndFeatures/azure-toolkit-for-intellij/idea/src/com/microsoft/azure/hdinsight/spark/actions/SparkSubmitJobAction.kt [hunk 2]
```diff
@@ -82,7 +82,8 @@ open class SparkSubmitJobAction : AzureAnAction() {
 
         runConfigurationSetting.isEditBeforeRun = true
 
-        val runConfiguration = runConfigurationSetting.configuration as? LivySparkBatchJobRunConfiguration ?: return
+        val runConfiguration = runConfigurationSetting.configuration as? LivySparkBatchJobRunConfiguration
```
Comment (maartenba) [inline PluginsAndFeatures/azure-toolkit-for-intellij/idea/src/com/microsoft/azure/hdinsight/spark/actions/SparkSubmitJobAction.kt]: Maybe keep as-is to make future merges easier?

**Review by maartenba — CHANGES_REQUESTED** [review]
Have not reviewed the Microsoft commits. Commented on a few small things, and also pushed new commits into your branch directly (so pull first).

Next to that: Listing of functions no longer seems to work?

Otherwise looks good to me!

**Review by maartenba — APPROVED** [review]

File: PluginsAndFeatures/azure-toolkit-for-intellij/idea/resources/META-INF/plugin.xml (hunk 1 above)
Comment (sdubov) [reply PluginsAndFeatures/azure-toolkit-for-intellij/idea/resources/META-INF/plugin.xml:34]: There is no icon in remote repo - https://github.com/Microsoft/azure-tools-for-java/blob/c9eab7bc014b0613a9b6b1233f925f09146973a1/PluginsAndFeatures/azure-toolkit-for-intellij/resources/META-INF/plugin.xml#L99

File: PluginsAndFeatures/azure-toolkit-for-intellij/idea/src/com/microsoft/azure/hdinsight/spark/actions/SparkSubmitJobAction.kt (hunk 2 above)
Comment (sdubov) [reply PluginsAndFeatures/azure-toolkit-for-intellij/idea/src/com/microsoft/azure/hdinsight/spark/actions/SparkSubmitJobAction.kt]: Fixed.

**Pull request merged**